runtime/compiler/p/codegen/PPCPrivateLinkage.cpp

```diff
@@ -1913,6 +1913,27 @@ int32_t TR::PPCPrivateLinkage::buildPrivateLinkageArgs(TR::Node
 	    addDependency(dependencies, NULL, (TR::RealRegister::RegNum)i, TR_VSX_SCALAR, cg());
       }
 
+   // All CHelper Direct Calls on platforms other than Linux 32bit must load the target Address, in this case the helper's address in r12 (Linux 64LE) or 
+   // load the address of the jitTOC into r2 (Linux 64BE, and AIX)
+   if(linkage == TR_CHelper)
+   {
+      if(TR::Compiler->target.isLinux() && TR::Compiler->target.is64Bit() && TR::Compiler->target.cpu.isLittleEndian())
+      {
+         uintptrj_t targetAddress = (uintptrj_t)callSymbol->getMethodAddress();
+         loadConstant(cg(), callNode, (int64_t)targetAddress,dependencies->searchPreConditionRegister(TR::RealRegister::gr12));
+      }
+      else if ((TR::Compiler->target.isAIX() || TR::Compiler->target.isLinux()) && TR::Compiler->target.is64Bit())
+      {
+         generateTrg1MemInstruction(cg(),TR::InstOpCode::Op_load, callNode, dependencies->searchPreConditionRegister(TR::RealRegister::gr2), 
+            new(trHeapMemory()) TR::MemoryReference(cg()->getMethodMetaDataRegister(), offsetof(J9VMThread, jitTOC), TR::Compiler->om.sizeofReferenceAddress(), cg()));
+      }
+      else if (TR::Compiler->target.isAIX())
```

Comment:
We can have just two cases, as @zl-wang summarized. Please include his summary as a comment.